eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1744,14 +1746,23 @@ protected function copyTranslationsFromPublishedVersion(APIVersionInfo $versionI
         $this->internalUpdateContent($versionInfo, $updateStruct);
     }
 
-    protected function isHashEqual(FieldType $fieldType, Value $newValue, Value $fieldValue): bool
+    protected function fieldValuesAreEqual(FieldType $fieldType, Value $value1, Value $value2): bool
     {
-        $newHash = $fieldType->toHash($newValue);
-        $currentHash = $fieldType->toHash($fieldValue);
-        if ($newValue instanceof \eZ\Publish\SPI\FieldType\Comparable) {
-            return $newValue->isEquals($newHash, $currentHash);
-        } elseif ($newHash === $currentHash) {
-            return true;
+        if ($fieldType instanceof Comparable) {
+            return $fieldType->valuesEqual($value1, $value2);
+        } else {
+            @trigger_error(
+                \sprintf(
+                    'In eZ Platform 2.5 and 3.x %s should implement %s. ' .
```

Comment:
Maybe instead of that we should think about moving `toHash` method to `\eZ\Publish\SPI\FieldType\Value`?